megaparse/core/megaparse.py

```diff
@@ -1,33 +1,47 @@
 import asyncio
 import os
 from pathlib import Path
+from typing import IO
 from langchain_openai import ChatOpenAI
 
 from megaparse.api.utils.type import FileExtension
 from megaparse.core.parser.unstructured_parser import UnstructuredParser
 
-from megaparse.core.parser import MegaParser
+from megaparse.core.parser import BaseParser
 from megaparse.core.checker.format_checker import FormatChecker
 
 
 class MegaParse:
     def __init__(
         self,
-        parser: MegaParser = UnstructuredParser(),
+        parser: BaseParser = UnstructuredParser(),
         format_checker: FormatChecker | None = None,
     ) -> None:
         self.parser = parser
         self.format_checker = format_checker
         self.last_parsed_document: str = ""
 
-    async def aload(self, file_path: Path | str) -> str:
-        if isinstance(file_path, str):
-            file_path = Path(file_path)
-        file_extension: str = file_path.suffix
+    async def aload(
+        self,
+        file_path: Path | str | None = None,
+        file: IO[bytes] | None = None,
+        file_extension: str | None = "",
+    ) -> str:
+        assert file_path or file, "Either file_path or file should be provided"
```

Comment:
not assert, but if not raise ValueError

assert to be used in units tests, not to catch errors